f8a_report/v2/report_generator.py

```diff
@@ -0,0 +1,488 @@
+                report_template['stacks_details'].append(stack_info_template)
+            except (IndexError, KeyError, TypeError) as e:
+                logger.exception('Error: %r' % e)
+                continue
+        logger.info("Stacks Analyse Completed.")
+        return report_template
+
+    def build_report_summary(self, unknown_deps_ingestion_report, report_content) -> dict:
+        """Build Final Report Summary."""
+        logger.info("Building Report summary.")
+
+        total_stack_requests = self.total_stack_requests
+        all_deps = self.all_deps
+        all_unknown_deps = self.all_unknown_deps
+        unique_stacks_with_recurrence_count = self.unique_stacks_with_recurrence_count
+        unique_stacks_with_deps_count = self.unique_stacks_with_deps_count
+        avg_response_time = self.avg_response_time
+        unknown_licenses = self.unknown_licenses
+        all_cve_list = self.all_cve_list
+        total_response_time = self.total_response_time
+        start_date = self.start_date
+        return {
+            'total_stack_requests_count': total_stack_requests['all'],
+            'npm': self.get_ecosystem_summary('npm', total_stack_requests, all_deps,
+                                              all_unknown_deps,
+                                              unique_stacks_with_recurrence_count,
+                                              unique_stacks_with_deps_count,
+                                              avg_response_time,
+                                              unknown_deps_ingestion_report),
+            'maven': self.get_ecosystem_summary('maven', total_stack_requests, all_deps,
+                                                all_unknown_deps,
+                                                unique_stacks_with_recurrence_count,
+                                                unique_stacks_with_deps_count,
+                                                avg_response_time,
+                                                unknown_deps_ingestion_report),
+            'pypi': self.get_ecosystem_summary('pypi', total_stack_requests, all_deps,
+                                               all_unknown_deps,
+                                               unique_stacks_with_recurrence_count,
+                                               unique_stacks_with_deps_count,
+                                               avg_response_time,
+                                               unknown_deps_ingestion_report),
+            'unique_unknown_licenses_with_frequency':
+                self.populate_key_count(unknown_licenses),
+            'unique_cves':
+                self.populate_key_count(all_cve_list),
+            'total_average_response_time':
+                '{} ms'.format(total_response_time['all'] / len(report_content['stacks_details'])),
+            'cve_report': CVE().generate_cve_report(updated_on=start_date)
+        }
+
+    def set_average_response_time(self) -> None:
```

Comment:
This Particular Function sets a class variable, So by default all Python functions return None.

Reply:
I have not understood, can you please explain why normalize_worker_data does not have a return type

Reply:
In case of Retraining (retrain=True)
`normalize_worker_data` -> Dict

Other Case: (retrain=False)
`normalize_worker_data` -> List

Their is change in return type in Trivial approach. I dont want to dive into writing whole new retraining functionality to match return type.

Reply:
Can you point me to some documentation with this syntax, I would like to understand it more.

Reply:
https://docs.python.org/3.6/library/typing.html

Reply:
Looks fine.